### PR TITLE
chore: adaptations for nightly-2024-02-15

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Testing/SlimCheck/Functions.lean [hunk 1]
```diff
@@ -7,6 +7,7 @@ import Mathlib.Data.List.Sigma
 import Mathlib.Data.Int.Range
 import Mathlib.Data.Finsupp.Defs
 import Mathlib.Data.Finsupp.ToDFinsupp
+import Std.Data.LazyList
 import Mathlib.Testing.SlimCheck.Sampleable
 import Mathlib.Testing.SlimCheck.Testable
 import Std.Data.List.Perm
```
Comment (Vierkantor) [inline Mathlib/Testing/SlimCheck/Functions.lean:13]: Let's keep these imports (almost) sorted.
```suggestion
import Mathlib.Data.Int.Range
import Mathlib.Data.Finsupp.Defs
import Mathlib.Data.Finsupp.ToDFinsupp
import Mathlib.Testing.SlimCheck.Sampleable
import Mathlib.Testing.SlimCheck.Testable
import Std.Data.LazyList
import Std.Data.List.Perm
```

File: Mathlib/Lean/Meta.lean [hunk 2]
```diff
@@ -3,10 +3,14 @@ Copyright (c) 2022 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro
 -/
+import Lean.Elab.Term
+import Lean.Elab.Tactic.Basic
```
Comment (Vierkantor) [inline Mathlib/Lean/Meta.lean:7]: More import sorting :P
```suggestion
import Lean.Elab.Tactic.Basic
import Lean.Elab.Term
```

File: Mathlib/Data/Set/Intervals/Basic.lean [hunk 3]
```diff
@@ -756,6 +756,10 @@ theorem Icc_self (a : α) : Icc a a = {a} :=
   Set.ext <| by simp [Icc, le_antisymm_iff, and_comm]
 #align set.Icc_self Set.Icc_self
 
+instance instIccUnique : Unique (Set.Icc a a) where
+  default := ⟨a, by simp⟩
+  uniq y := Subtype.ext <| by simpa using y.2
+
```
Comment (Vierkantor) [inline Mathlib/Data/Set/Intervals/Basic.lean:762]: This came from merging master into this branch. Hopefully that doesn't mess up the version history.

**Comment by jcommelin** [comment]
bors r-

I see that @Vierkantor made some good points

**Comment by semorrison** [comment]
bors p=10

**Comment by semorrison** [comment]
bors merge

**Pull request merged**